src/main/java/hudson/plugins/chucknorris/CordellWalkerRecorder.java

```diff
@@ -111,17 +122,52 @@ public final Action getProjectAction(final AbstractProject<?, ?> project) {
     public final boolean perform(final AbstractBuild<?, ?> build,
             final Launcher launcher, final BuildListener listener)
             throws InterruptedException, IOException {
-        Style style = Style.get(build.getResult());
-        String fact = factGenerator.random();
-        build.getActions().add(new RoundhouseAction(style, fact));
+        perform(build);
         return true;
     }
 
     /**
+     * Adds RoundhouseAction to the run actions. This is applicable for each
+     * run.
+     * @param run
+     *            the run
+     * @param workspace
+     *            the workspace
+     * @param launcher
+     *            the launcher
+     * @param listener
+     *            the listener
+     * @throws InterruptedException
+     *             when there's an interruption
+     * @throws IOException
+     *             when there's an IO error
+     */
+    @Override
+    public final void perform(
+            @Nonnull final Run<?, ?> run, @Nonnull final FilePath workspace,
+            @Nonnull final Launcher launcher, @Nonnull final TaskListener listener)
+            throws InterruptedException, IOException {
+        perform(run);
+    }
+
+    /**
      * Gets the required monitor service.
      * @return the BuildStepMonitor
      */
     public final BuildStepMonitor getRequiredMonitorService() {
         return BuildStepMonitor.NONE;
     }
+
+    /**
+     * Adds RoundhouseAction to the run actions. This is applicable for each
+     * run.
+     * @param run
+     *            the run
+     */
+    private void perform(final Run<?, ?> run) {
+        Style style = Style.get(run.getResult());
+        String fact = factGenerator.random();
+        run.addAction(new RoundhouseAction(style, fact));
+    }
+
```

Comment:
If you add a descriptor here with an `@Symbol` (you'll have to add a dependency for structs) you could use shorthand syntax instead of `step([$class:...])`